src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -554,10 +556,14 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
     }
 
 
-    private String[] getPreRunnerSwitches() {
+    private String[] getPreRunnerSwitches() throws MatlabVersionNotFoundException {
         String[] preRunnerSwitches =
                 {getLocalMatlab() + nodeSpecificfileSeparator + "bin" + nodeSpecificfileSeparator + "matlab", "-nosplash",
-                        "-nodesktop", "-noAppIcon"};
+                        "-nodesktop"};
+        MatlabReleaseInfo rel = new MatlabReleaseInfo(this.nodeSpecificMatlabRoot);
+        if(!rel.verLessThan(MatlabBuilderConstants.BASE_MATLAB_VERSION_NO_APP_ICON_SUPPORT)) {
+        	preRunnerSwitches =  (String[]) ArrayUtils.add(preRunnerSwitches, "-noAppIcon");
+        } 
```

Comment:
As far as I know, VersionInfo.xml did not exist in 8.6 (R2015b). Are you going to update MatlabReleaseInfo to examine Content.m for the version in the event VersionInfo.xml is not available? Or are you going to wait until you switch over to run_matlab_command?

Reply:
Makes sense, I  by default send  VERSION_16B (9.1) if i dont get VersionInfo. I could adopt Content.m and update the MatlabReleaseInfo  however i see the point for waiting till we adopt run_matlab_command. I feel it only depends on how early we could release the changes and how important this fix is. I will have the MatlabRelease Info updated with Content.m logic. We can disscuss about it.